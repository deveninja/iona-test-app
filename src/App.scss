body {
  background-color: bisque;
  font-family: 'Gluten', cursive;
  color:dimgray;

  a {
    color: hsl(240, 38%, 41%);
  }
  
}