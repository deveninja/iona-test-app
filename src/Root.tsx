import { Provider } from 'react-redux';
import Store from 'store';

/**
 * @description - This will be used to improve testing suites and
 * as single source of provider wrapper
 * @param param0 - children props
 * @returns Provider wrapper for all children
 */
const Root: React.FC = ({ children }) => {
    return (
        <Provider store={Store}>
            {children}
        </Provider>
    )
}

export default Root