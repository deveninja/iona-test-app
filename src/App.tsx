import LazyLoad from 'helpers/LazyLoader'
import { AppRoute } from 'enums/route'
import Main from 'pages/main/Main'
import { Switch, Route, BrowserRouter } from 'react-router-dom'
import './App.scss'
import Notification from 'components/common/notification/Notification'

const CatsPage = LazyLoad(() => import('pages/cats/CatsPage').then(module => module.default))
const HomePage = LazyLoad(() => import('pages/home/Home').then(module => module.default))

const App: React.FC = () => {

  return (
    <Main>
      <Notification />

      <BrowserRouter>
        <Switch>
          <Route exact path={AppRoute.Home} component={HomePage} />
          <Route path={AppRoute.Cats} component={CatsPage} />
        </Switch>
      </BrowserRouter>
    </Main>
  )
}

export default App;
