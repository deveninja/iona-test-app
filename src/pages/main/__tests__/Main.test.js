import renderer from 'react-test-renderer'
import App from 'App'
import Root from 'Root'
import { cleanup, render } from '@testing-library/react'



describe('<Main />', () => {

  let app

  beforeEach(() => {
    app = render(<Root><App /></Root>)
  })

  afterEach(() => cleanup())

  it('displays the main section', () => {
    const mainSection = app.getByTestId('mainSection')
    expect(mainSection).toBeTruthy()
  })

  it('matches snapshot', () => {
    const appSnapshot = renderer.create(<Root><App /></Root>).toJSON()
    expect(appSnapshot).toMatchSnapshot()
  })
})
