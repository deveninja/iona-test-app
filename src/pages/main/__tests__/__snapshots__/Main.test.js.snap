// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> matches snapshot 1`] = `
<div
  data-testid="mainSection"
>
  <div
    className="toast-container position-absolute top-0 end-0 p-3"
    style={
      Object {
        "color": "white",
        "zIndex": 2,
      }
    }
  />
  <div
    className="home container"
  >
    <div
      className="cardSection"
      data-testid="cardSection"
    >
      <div
        className="row row"
      >
        <h1
          data-testid="pageHeading"
        >
          Select Species
        </h1>
        <div
          className="card col-lg"
        >
          <div
            className="card card"
            style={
              Object {
                "width": undefined,
              }
            }
          >
            <a
              href="/cats"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="imageWrapper"
              >
                <img
                  alt="pet"
                  src="https://i.pinimg.com/originals/83/01/aa/8301aa7d789494566ea9ea1e229c5a36.jpg"
                />
              </div>
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Feline
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="card col-lg"
        >
          <div
            className="card card"
            style={
              Object {
                "width": undefined,
              }
            }
          >
            <a
              href="/dogs"
              onClick={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="imageWrapper"
              >
                <img
                  alt="pet"
                  src="https://i.pinimg.com/originals/de/41/7f/de417fa33ab730fab446d44b7749703d.jpg"
                />
              </div>
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Canine
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
